src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -201,151 +201,160 @@ public boolean canBeRotated() {
-                    itemOut = itemIn.copy();
-                    itemIn = null;
-                } else {
-                    itemOut = itemIn.copy();
-
-                    itemOut.stackSize = 1;
-                    itemIn.stackSize = itemIn.stackSize - 1;
-                }
-
-                if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
-                if (itemOut.stackSize == 0) itemOut = null;
-
-                inventory.setInventorySlotContents(0, itemIn);
-                inventory.setInventorySlotContents(1, itemOut);
-
-                ticksRemaining = 200;
-                machineActive = true;
-
-                this.markForUpdate();
-                this.markDirty();
+        if (fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0) {
+            if (inventory.getStackInSlot(1) != null) {
+                burnTime(1);
+            }
+            if (inventory.getStackInSlot(0) != null) {
+                burnTime(0);
             }
+        }
+
 
+        if (this.canWork() && inventory.getStackInSlot(0) != null && inventory.getStackInSlot(1) == null) {
+            if (!cancrush(inventory.getStackInSlot(0)))
+                return;
 
-            if (fuelRemaining == 0 && machineActive && machineTier == machineTier.TIER_0) {
-                ticksRemaining = 0;
-                test = false;
+            ItemStack itemIn = inventory.getStackInSlot(0);
+            ItemStack itemOut;
 
+            if (itemIn.stackSize - 1 <= 0) {
+                itemOut = itemIn.copy();
+                itemIn = null;
             } else {
-                test = false;
+                itemOut = itemIn.copy();
+                itemOut.stackSize = 1;
+                itemIn.stackSize = itemIn.stackSize - 1;
             }
+            if (itemIn != null && itemIn.stackSize == 0) itemIn = null;
+            if (itemOut.stackSize == 0) itemOut = null;
 
+            inventory.setInventorySlotContents(0, itemIn);
+            inventory.setInventorySlotContents(1, itemOut);
 
-            if (ticksRemaining <= 0 && machineActive) {
-
-                if (test == false) {
-                    System.out.println(test);
-                    ticksRemaining = 0;
-
-                    if (worldObj.isRemote)
-                        return;
+            ticksRemaining = 200;
+            machineActive = true;
+            this.markForUpdate();
+            this.markDirty();
+        }
 
-                    ItemStack processItem = inventory.getStackInSlot(1);
-                    if (processItem == null) {
-                        return;
-                    }
+        if (inventory.getStackInSlot(1) != null && ticksRemaining <= 0) {
 
-                    List<Crushable> outputs = PulverizerRegistry.getOutputs(processItem);
+            ticksRemaining = 0;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS106)